src/v/pandaproxy/schema_registry/handlers.cc

```diff
@@ -523,4 +524,35 @@ status_ready(server::request_t rq, server::reply_t rp) {
     co_return rp;
 }
 
+ss::future<server::reply_t>
+put_admin_redpanda_services_restart(server::request_t rq, server::reply_t rp) {
+    parse_accept_header(rq, rp);
+    auto _service{parse::query_param<rp_service>(*rq.req, "service")};
+    rq.req.reset();
+
+    if (_service.kind != service_kind::schema_registry) {
+        ss::sstring msg = ssx::sformat("Invalid RP service: {}", _service.kind);
+        vlog(plog.error, "{}", msg);
+        rp.rep = errored_body(reply_error_code::not_found, msg);
+        co_return rp;
+    }
+
+    vlog(plog.info, "restart internal store");
+
+    auto client_cfg = config::to_yaml(
+      rq.service().client().local().config(), config::redact_secrets::no);
+    co_await rq.service().client().stop();
+    co_await rq.service().writer().clear();
+    co_await rq.service().client().invoke_on_all(
+      [](kafka::client::client& client) { return client.clear(); });
+    co_await rq.service().schema_store().clear();
+    co_await rq.service().client().start(
+      client_cfg, [&rq](std::exception_ptr ex) {
+          return rq.service().mitigate_error(ex);
+      });
```

Comment:
> I need to see how to retain the same `external_mitigate` that is originally set in `src/v/pandaproxy/schema_registry/api.cc`:
> ```cpp
>    co_await _client.start(
>      config::to_yaml(_client_cfg, config::redact_secrets::no),
>      [this](std::exception_ptr ex) {
>          return _service.local().mitigate_error(ex);
>      });
> ```

@BenPope Is this safe? I believe the `rq` object is destroyed after control leaves this scope.

Reply:
`rq` is destroyed, but the reference returned from `rq.service()` should be stable.